mirrord-layer/src/sockets.rs

```diff
@@ -193,12 +193,12 @@ fn bind(sockfd: c_int, addr: *const sockaddr, addrlen: socklen_t) -> c_int {
         return unsafe { libc::bind(sockfd, addr, addrlen) };
     }
 
-    socket.state = SocketState::Bound(Bound {
+    Arc::get_mut(&mut socket).unwrap().state = SocketState::Bound(Bound {
```

Comment:
`socket.get_mut` instead of calling the fully qualified name

Reply:
the get_mut method is only available as an associated function, so we have to use it like so. Otherwise I can wrap the socket in a Pin and then use it